components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -1104,6 +1131,28 @@ public void revokeAccessToken(String tokenId, String userId) throws IdentityOAut
         return accessTokens;
     }
 
+    /**
+     * Checks whether id_tokens are issued for application tokens (ie. tokens issued for client_credentials grant type)
+     *
+     * @return
+     */
+    private boolean isIdTokenIssuedForApplicationTokens() {
+
+        return !OAuthServerConfiguration.getInstance().getIdTokenNotAllowedGrantTypesSet()
```

Comment:
getIdTokenNotAllowedGrantTypesSet can be null.

Reply:
getIdTokenNotAllowedGrantTypesSet() returns an empty set if there aren't any values

Reply:
+1